core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -454,7 +454,14 @@ class SparkContext(config: SparkConf) extends Logging {
 
     // Set Spark driver host and port system properties. This explicitly sets the configuration
     // instead of relying on the default value of the config constant.
-    _conf.set(DRIVER_HOST_ADDRESS, _conf.get(DRIVER_HOST_ADDRESS))
+    if (master.startsWith("k8s") &&
+      _conf.getBoolean("spark.kubernetes.executor.useDriverPodIP", defaultValue = false)) {
```

Comment:
Indentation? Two more spaces.

Reply:
As a side node, although we recommend to mention `defaultValue =`, let's simplify like the following like the other places of `getBoolean` in `SparkContext` file.
```scala
_conf.getBoolean("spark.kubernetes.executor.useDriverPodIP", defaultValue = false)) {
_conf.getBoolean("spark.kubernetes.executor.useDriverPodIP", false)) {
```